src/h5cpp/node/functions.cpp

```diff
@@ -201,26 +203,24 @@ void link(const Node &target,
   }
   else
   {
-    auto base = link_base;
-    auto lpath = link_path.name();
-    if (link_path.absolute())
-    {
-      base = link_base.link().file().root();
-      auto stripped = link_path;
-      stripped.absolute(false);
-      lpath = static_cast<std::string>(stripped);
-    }
-    if (base.links.exists(lpath))
+    //we are in the same file - can create a symbolic link
```

Comment:
Nicely done. Thanks for cleaning this up. But what if supplied link_path is absolute? I see two options: A) throw or B) ignore the fact that it is absolute, assume that relative was meant regardless, or C) create link at absolute path of the file within which link_base is. Furthermore, your check for whether he base node is a group is quite nice, but are we sure the C API does not already throw in such a case?

Reply:
The key here is the get_node function which basically manages all the complexity. If the path is relative we know what to do. In the case of an absolute path, well, we use base to determine the root group and than happily resolve the absolute path.

Reply:
Concerning the iterators: 
* they traverse depth first. I would leave it this way as it would make the implementation far more   complex and there is no real benefit from it. If you wand to have elements in a particular order copy them in vector and sort them in whatever fashion you want. 
* there is a problem with cycles. I have no idea how to prevent this but I will definitely mention it in the documentation.